adapters/repos/db/lsmkv/proxy_bucket_test.go

```diff
@@ -0,0 +1,252 @@
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package lsmkv
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/weaviate/weaviate/adapters/repos/db/helpers"
+	"github.com/weaviate/weaviate/adapters/repos/db/inverted/tracker"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+)
+
+func TestBucketProxyCreation(t *testing.T) {
+	tmpDir := "/tmp/lsmkvtestdir"
+	os.Mkdir(tmpDir, 0o777)
+	ctx := context.Background()
+	logger, _ := test.NewNullLogger()
+	propName := "testPropertyName"
+	propids, err := tracker.NewJsonPropertyIdTracker(tmpDir + "/propids.json")
+	if err != nil {
+		t.Fatalf("Failed to create tracker: %v", err)
+	}
+
+	b, err := NewBucket(ctx, tmpDir, "", logger, nil, cyclemanager.NewCallbackGroupNoop(), cyclemanager.NewCallbackGroupNoop())
+	if err != nil {
+		t.Fatalf("Failed to create bucket: %v", err)
+	}
+
+	bp, err := NewBucketProxy(b, propName, propids)
+	if err != nil {
+		t.Fatalf("Failed to create BucketProxy: %v", err)
+	}
+
+	if bp == nil {
+		t.Fatal("Failed to create BucketProxy")
+	}
+
+	propid_bytes, err := helpers.MakePropertyPrefix(propName, propids)
+	if err != nil {
+		t.Fatalf("Failed to create property prefix: %v", err)
+	}
+	if !bytes.Equal(bp.PropertyPrefix(), propid_bytes) {
+		t.Fatalf("BucketProxy PropertyPrefix() does not match expected '%s', got '%s'", propName, bp.PropertyPrefix())
+	}
+}
+
+func TestBucketProxyGetAndPut(t *testing.T) {
+	tmpDir := "/tmp/lsmkvtestdir"
+	os.Mkdir(tmpDir, 0o777)
+	ctx := context.Background()
+	logger, _ := test.NewNullLogger()
+	propName := "testPropertyName"
+	propids, err := tracker.NewJsonPropertyIdTracker(tmpDir + "/propids.json")
+	if err != nil {
+		t.Fatalf("Failed to create tracker: %v", err)
+	}
+
+	b, err := NewBucket(ctx, tmpDir, "", logger, nil, cyclemanager.NewCallbackGroupNoop(), cyclemanager.NewCallbackGroupNoop())
+	if err != nil {
+		t.Fatalf("Failed to create bucket: %v", err)
+	}
+
+	bp, err := NewBucketProxy(b, propName, propids)
+	if err != nil {
+		t.Fatalf("Failed to create BucketProxy: %v", err)
+	}
+
+	// Put a value into the BucketProxy
+	key := []byte("testKey")
+	value := []byte("testValue")
+	err = bp.Put(key, value)
+	if err != nil {
+		t.Fatalf("Failed to put value into BucketProxy: %v", err)
+	}
+
+	// Retrieve the value from the BucketProxy
+	retrieved, err := bp.Get(key)
+	if err != nil {
+		t.Fatalf("Failed to get value from BucketProxy: %v", err)
+	}
+
+	// Check that the retrieved value matches the put value
+	if string(retrieved) != string(value) {
+		t.Fatalf("Retrieved value does not match put value: expected '%s', got '%s'", string(value), string(retrieved))
+	}
+}
+
+func TestBucketProxyGetNotFound(t *testing.T) {
+	tmpDir := "/tmp/lsmkvtestdir"
+	os.Mkdir(tmpDir, 0o777)
```

Comment:
golang tests have a built-in method for creating temporary directories `dirName := t.TempDir()` please use it everywhere you make a temp dir instead of manual folder creation. `t.TempDir()` also cleans itself after the tests are run